contracts/virus/src/contract.rs

```diff
@@ -60,9 +60,11 @@ pub fn execute_spread(
 
         attributes.push(Attribute::new(format!("path{i}"), path.clone()));
 
+        // clippy produces a false positive here, see https://github.com/rust-lang/rust-clippy/issues/9841
+        #[allow(clippy::explicit_auto_deref)]
         let address = deps
             .api
-            .addr_humanize(&instantiate2_address(&checksum, &creator, &salt)?)?;
+            .addr_humanize(&instantiate2_address(&*checksum, &creator, &salt)?)?;
```

Comment:
Maybe we should have an `checksum.as_slice()` or something like that

Reply:
This case can be solved with `checksum.as_ref()`, since the array reference it returns will be coerced to a slice, but we can also add an `as_slice` to get it directly.